src/Google.Api.Gax/Api/Gax/ApiCallableExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Protobuf;
+using Grpc.Core;
+using System.Threading;
+
+namespace Google.Api.Gax
+{
+    /// <summary>
+    /// Extension methods on <see cref="ApiCallable{TRequest, TResponse}"/>.
+    /// </summary>
+    public static class ApiCallableExtensions
+    {
+        /// <summary>
+        /// Creates a retrying equivalent of the given callable delegate.
+        /// </summary>
+        /// <typeparam name="TRequest">Request message type</typeparam>
+        /// <typeparam name="TResponse">Response message type</typeparam>
+        /// <param name="originalCall">Original callable delegate</param>
+        /// <param name="settings">Retry settings</param>
+        /// <param name="clock">Clock to consult for retry timing</param>
+        /// <returns>A retrying equivalent of <paramref name="originalCall"/></returns>
+        public static ApiCallable<TRequest, TResponse> WithRetry<TRequest, TResponse>(
+            this ApiCallable<TRequest, TResponse> originalCall,
+            RetrySettings settings,
+            IClock clock)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse> => WithRetry(originalCall, settings, clock, null);
+
+        /// <summary>
+        /// Creates a retrying equivalent of the given callable delegate.
+        /// </summary>
+        /// <typeparam name="TRequest">Request message type</typeparam>
+        /// <typeparam name="TResponse">Response message type</typeparam>
+        /// <param name="originalCall">Original callable delegate</param>
+        /// <param name="settings">Retry settings</param>
+        /// <param name="clock">Clock to consult for retry timing</param>
+        /// <param name="scheduler">Scheduler to use for delays</param>
+        /// <returns>A retrying equivalent of <paramref name="originalCall"/></returns>
+        public static ApiCallable<TRequest, TResponse> WithRetry<TRequest, TResponse>(
+            this ApiCallable<TRequest, TResponse> originalCall,
+            RetrySettings settings,
+            IClock clock,
+            IScheduler scheduler)
+            where TRequest : class, IMessage<TRequest>
+            where TResponse : class, IMessage<TResponse>
+        {
+            GaxPreconditions.CheckNotNull(originalCall, nameof(originalCall));
+            GaxPreconditions.CheckNotNull(clock, nameof(clock));
+            scheduler = scheduler ?? SystemScheduler.Instance;
+            GaxPreconditions.CheckNotNull(settings, nameof(settings));
```

Comment:
Put this precondition check with the others? And in the same order as the parameters?